pyrsa/data/noise.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Apr 13 12:42:06 2020
+
+@author: heiko
+"""
+
+from collections.abc import Iterable
+import numpy as np
+
+
+def cov_from_residuals(residuals, dof=None):
+    """
+    computes an optimal shrinkage estimate of the precision matrix from 
+    the residuals as described by Ledoit and Wolfe (2004): "A well-conditioned
+    estimator for large-dimensional covariance matrices"
+
+    Args:
+        residuals(numpy.ndarray or list of these): n_obs x n_channels matrix
```

Comment:
please rename n_obs, such that it does not get confused with n_obs in dataset/rdm objects